### PR TITLE
Release 2.0

### PR DESCRIPTION
**Pull request by pirj**
Prepare for a 2.0 release.

This PR is to track the changes in `release-2.0` branch, which is purposed to integrate [remaining PRs targeted for 2.0](https://github.com/rubocop-hq/rubocop-rspec/pulls?q=is%3Aopen+is%3Apr+milestone%3A2.0).

### CODE REVIEW

File: lib/rubocop/rspec/version.rb [hunk 1]
```diff
@@ -4,7 +4,7 @@ module RuboCop
   module RSpec
     # Version information for the RSpec RuboCop plugin.
     module Version
-      STRING = '1.44.1'
+      STRING = '2.0.0.pre'
```
Comment (pirj) [inline lib/rubocop/rspec/version.rb:7]: 2.0.0

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,13 @@
 
 ## Master (Unreleased)
 
+* Remove deprecated class `::RuboCop::Cop::RSpec::Cop`. ([@bquorning][])
+
+## 2.0.0.pre (2020-10-22)
```
Comment (bquorning) [inline CHANGELOG.md:7]:
```suggestion
## 2.0.0 (2020-10-?)
```

**Comment by bquorning** [comment]
I have taken a look at the remaining pull requests before 2.0, and tried to put them in order:

1. #942: Add mixins folder (Can be done before or after #969)
2. #956: RSpec aliases runtime matchers – needs to be done before
3. #1007: Drop RSpec/EmptyExampleGroup customization (Blocked by #956)
4. #967: Optimize Base#relevant_rubocop_rspec_file? (Blocked by #956)
5. #1013: Add v2 update docs 
6. #1055: Release 2.0

**Comment by pirj** [comment]
Awesome!

I can take care of making final touches to #956. I really hope it doesn't interfere with #942 :scream:

**Comment by bquorning** [comment]
Would you like me to take a stab at rebasing #942?

**Comment by bquorning** [comment]
> I can take care of making final touches to #956. I really hope it doesn't interfere with #942 😱

It looks like there will only be conflicts in lib/rubocop/rspec/top_level_group.rb and lib/rubocop/rspec/variable.rb. It looks like minor stuff. I think it’s still worth it to work in parallel. You take #956, I take #942.

**Comment by pirj** [comment]
Sounds good :+1:

**Comment by bquorning** [comment]
@pirj I wish GitHub offered a way of communicating 1:1, synchronously 😄 I’ll try signing onto gitter.im, you can catch me there if you want.

**Comment by bquorning** [comment]
I added a commit to bump the version number to 2.0.0. Are we ready? Are there any easy-peasy bugfix PRs that we want to also merge in?

**Comment by pirj** [comment]
> Should we link to @pirj ’s upgrade document

Great idea!

> Are we ready?

YES! 🚀  🙌

**Comment by Darhazer** [comment]
I guess [this](https://github.com/rubocop-hq/rubocop-rspec/pull/1075#discussion_r518034541) should be fixed first

File: rubocop-rspec.gemspec [hunk 3]
```diff
@@ -36,8 +36,7 @@ Gem::Specification.new do |spec|
     'documentation_uri' => 'https://docs.rubocop.org/rubocop-rspec/'
   }
 
-  spec.add_runtime_dependency 'rubocop', '~> 0.87'
-  spec.add_runtime_dependency 'rubocop-ast', '>= 0.7.1'
+  spec.add_runtime_dependency 'rubocop', '~> 1.0'
```
Comment (pirj) [inline rubocop-rspec.gemspec:39]: I believe we also need `rubocop-ast` 1.1.0 for DSL aliases's `Module.method` syntax. @bquorning

Comment (pirj) [reply rubocop-rspec.gemspec:39]: I'm sorry I've overlooked this in #956

Comment (bquorning) [reply rubocop-rspec.gemspec:39]: Hmm, which of these, then?

1. `'>= 1.1.0'`
2. `'~> 1.1.0'`
3. `'~> 1.1'`

I think 3 should be fine, right?

Comment (pirj) [reply rubocop-rspec.gemspec:39]: 1 is the best.

**Review by bquorning — APPROVED** [review]

**Comment by pirj** [comment]
@Darhazer, green light for the release?

**Comment by pirj** [comment]
Hey @backus!

Summoning you not only to notify you about an upcoming major version release of a child of yours.
We seem to lack CodeClimate keys, do you mind to [invite](https://docs.codeclimate.com/docs/organization-roles-permissions) @bquorning, @Darhazer and myself @pirj [there](https://codeclimate.com/github/rubocop-hq/rubocop-rspec/) as collaborators?

**Comment by Darhazer** [comment]
> @Darhazer, green light for the release?

I trust you @pirj and @bquorning on this one. Release at your will and thank you for the hard work to make it happen :tada:

**Comment by bquorning** [comment]
- [Gem](https://rubygems.org/gems/rubocop-rspec/versions/2.0.0)
- [GH release](https://github.com/rubocop-hq/rubocop-rspec/releases/v2.0.0)
- [Docs](https://docs.rubocop.org/rubocop-rspec/2.0/index.html)

Did I miss anything? Except saying a bit **thank you to everyone** who helped getting us this far.

**Comment by pirj** [comment]
Awesome! Congrats with the release, guys!

**Comment by pirj** [comment]
Closed a dozen tickets, mostly thanks to #956 🎉

**Pull request merged**